src/DGtal/kernel/UnorderedSetByBlock.h

```diff
@@ -0,0 +1,1218 @@
+    typedef Splitter< Element >  Self;
+    typedef uint32_t             Word;
+
+    /// Splits an element \a e into a pair grouping its block
+    /// coordinates and its bit within this block.
+    ///
+    /// @param e any lattice point
+    ///
+    /// @return a pair grouping its block coordinates and its bit
+    /// within this block.
+    static
+    std::pair< Element, DGtal::Dimension >
+    split( const Element& e )
+    {
+      Element se = e;
+      se[ 0 ]   &= 0xffffffe0;
+      return std::make_pair( se, (DGtal::Dimension) e[ 0 ] & 0x0000001f );
+    }
+
+    /// Rejoins a splitted element (see \ref split).
+    ///
+    /// @param e the block coordinate of the element.
+    /// @param x the bit coordinate of the element.
+    ///
+    /// @return the corresponding rejoined element.
+    static
+    Element
+    join( const Element& e, DGtal::Dimension x )
+    {
+      Element ge = e;
+      ge[ 0 ] |= x;
+      return ge;
+    }
+
+    /// Rejoins a splitted element (see \ref split).
+    ///
+    /// @param p a pair grouping the block coordinate and the bit
+    /// coordinate of the element.
+    ///
+    /// @return the corresponding rejoined element.
+    static
+    Element
+    join( const std::pair< Element, DGtal::Dimension >& p )
+    {
+      Element ge = p.first;
+      ge[ 0 ] |= p.second; 
+      return ge;
+    }
+  }; // struct Splitter< PointVector< dim, DGtal::int32_t, TContainer > >
+
+  /// Splits an integral-array element into an integral-array element
+  /// and a 32 bit integer. The expected behaviour is for an element e
+  /// of dimension 3:
+  /// \code
+  /// Splitter< Point > split;
+  /// auto v = split( Point( 117, 43, 52 ) );
+  /// v.first  == 117 & 0x1f
+  /// v.second == Point( 117 & 0xffff...ffe0, 43, 52 );
+  /// \endcode
+  ///
+  /// @tparam dim the dimension of the integral array.
+  /// @tparam TContainer the container for PointVector.
+  ///
+  /// @note In subclass, we use mask operations instead of mult/div 32. The result
+  /// 4 x times faster !
+  ///
+  /// @note Specialization for 32 bits integer types
+  ///
+  /// @see UnorderedSetByBlock
+  template < DGtal::Dimension dim, typename TContainer >
+  struct Splitter< PointVector< dim, DGtal::int64_t, TContainer > > {
+    typedef PointVector< dim, DGtal::int64_t, TContainer > Element;
+    typedef Splitter< Element >  Self;
+    typedef uint32_t             Word;
+
+    /// Splits an element \a e into a pair grouping its block
+    /// coordinates and its bit within this block.
+    ///
+    /// @param e any lattice point
+    ///
+    /// @return a pair grouping its block coordinates and its bit
+    /// within this block.
+    static
+    std::pair< Element, DGtal::Dimension >
+    split( const Element& e )
+    {
+      Element se = e;
+      se[ 0 ]   &= 0xffffffffffffffe0ll;
+      return std::make_pair( se, (DGtal::Dimension) e[ 0 ] & 0x000000000000001fll );
```

Comment:
I was more confident with bitwise operations. I thought /32 and %32 would work fine: they did but were 4 times slower. Do you think your representation is clearer ?
Otherwise, yes 64 bit word could be considered and is would be cheaper on average. But I was lazy I guess. In a forthcoming PR perhaps, because it would require some benchmarking probably ?

Reply:
It is not about the bitwise operations but specifically mixing signed and unsigned integers, like the `0xffffffffffffffe0` mask stored in a signed integer. I completely agree that using integer division and modulo will be slower and, BTW, will not handle negative coordinates appropriately (e.g. -40 will be mapped to the -32 block with -8 coordinate) without additional test.

I was more thinking to the version that uses the decimal representation and a subtraction (instead of a mask) to get the inside coordinates, e.g.:
```C++
static
std::pair< Element, DGtal::Dimension >
split( Element e )
{
    auto block_coords = static_cast<DGtal::Dimension>(e[ 0 ] & static_cast<typename Element::Coordinate>(31));
    e[ 0 ] -= block_coords;    
    return { e, block_coords };
}
```

I reused `Element` that is passed by copy instead of reference. Maybe to much `static_cast` but it is to avoid unwanted implicit conversions (and associated warnings). The advantages are that the block size is thus more easily customizable (for later, e.g. using `std::bitset` that should be equivalent to using `uintN_t` for N <= 64) and there is not need to have two specialized versions (still need to detect if a fundamental type is used or note :thinking: )

Reply:
Ok. I see your point. I modify the code and also allows the user to choose word type